Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -945,34 +941,71 @@ def kill_zombie_processes():
         demisto.debug(f"Failed to iterate over processes. Error: {e}")
 
 
+def extract_hostname(url: str) -> str:
+    """
+    Extract hostname from URL, adding http:// if protocol is missing.
+
+    Args:
+        url (str): The URL to process
+
+    Returns:
+        str: The extracted hostname
+    """
+    if not url.startswith(("http://", "https://")):
+        url = "http://" + url
+    try:
+        parsed = urlparse(url)
+        return parsed.netloc.split(":")[0]  # Remove port if exists
+    except Exception:
+        return ""
+
 @lru_cache(maxsize=1024)
 def is_private_network(url: str) -> bool:
     """
     Check if a URL's hostname belongs to a private network.
-    
+
     Args:
         url (str): The URL to check
-        
+
     Returns:
         bool: True if the hostname is in a private network, False otherwise
     """
     try:
-        if not url.startswith(('http://', 'https://')):
-            url = 'http://' + url
-        parsed = urlparse(url)
-        hostname = parsed.netloc.split(':')[0]  # Remove port if exists
-        
-        if not hostname:  # Handle cases where netloc is empty
+        if not (hostname := extract_hostname(url)):
             return False
             
-        ip = ipaddress.ip_address(hostname)
-        return any(ip in network for network in PRIVATE_NETWORKS)
-            
+        return ipaddress.ip_address(hostname).is_private
+        
     except (ValueError, AttributeError):
-        # ValueError: Invalid IP address
-        # AttributeError: Invalid URL format
         return False
 
+def remove_leading_zero_from_ip_addresses(path: str) -> str:
+    """
+    Removes leading zeros from IP addresses in the given path.
+    as leading zeros is not valid in IP addresses.
+    This function will only remove leading zeros from the IP address
+    Args:
+        path (str): The path to process.
+
+    Returns:
+        str: The processed path with leading zeros removed from IP addresses.
+    """
+    if not (hostname := extract_hostname(path)):
+        return path
+
+    # If hostname contains letters, it's not an IP address
+    if bool(re.search('[a-zA-Z]', hostname)):
+        return path
+    # Check if the hostname is an IP address
+    # Check if it's a valid IP address
+    ip_pattern = r'^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
+    if re.match(ip_pattern, hostname):
+        octets = hostname.split('.')
+        normalized_ip = '.'.join(str(int(octet)) for octet in octets)
+        return path.replace(hostname, normalized_ip)
+        
+    return path
```

Comment:
Add an info-level log message if the return value is different from the input value.